File: python/sortingalgorithm/bubblesort.py
def bubbleSort(array):
    
    length = len(array)

    for i in range(length - 1, 0, -1):
        for n in range(i):
            if array[n] > array[n+1]:
                array[n], array[n+1] = array[n+1], array[n]
                print(array)
            else:
                return

array = [4, 3, 2, 1]
print("Pre-run list is: ", array)
bubbleSort(array)
print("Post-run list is: ", array)

File: java/sortingalgorithm/bubblesort.java
public class bubblesort {
	public static void main(String args[])
	{
		int[] array = {3, 6, 8, 1, 2, 4};


		for (int n = array.length - 2; n > 0; n--) {
			for (int i = 0; i < n; i++) {
				if (array[n] > array[n+1]) {
					int temp = array[i];
					array[i] = array[i+1];
					array[i+1] = temp;
				}
			}
		}

		for (int i : array) {
			System.out.print(i + " ");
		}

	}
}
